include/seastar/net/tcp.hh

```diff
@@ -2089,8 +2087,14 @@ tcp_seq tcp<InetTraits>::tcb::get_isn() {
     hash[0] = _local_ip.ip;
     hash[1] = _foreign_ip.ip;
     hash[2] = (_local_port << 16) + _foreign_port;
-    hash[3] = _isn_secret.key[15];
-    CryptoPP::Weak::MD5::Transform(hash, _isn_secret.key);
+    gnutls_hash_hd_t md5_hash_handle;
+    // GnuTLS digests do not init at all, so this should never fail.
+    gnutls_hash_init(&md5_hash_handle, GNUTLS_DIG_MD5);
```

Comment:
What about the memory allocation you mentioned?

Reply:
what you mean by "what about"? are you asking on the overhead incurred by the allocation/deallocation ?

Reply:
@avikivity ping

Reply:
@avikivity i was trying to explain that `gnutls_has_init()` does not incur any performance regression which one might suspects that this could introduce due to the dynamic memory management. but i audited the code of GnuTLS when implementing this change and came to the conclusion.